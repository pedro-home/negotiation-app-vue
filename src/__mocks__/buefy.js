import Buefy from 'buefy'

export default {
  ...Buefy,
  install: function(Vue, config) {
    Buefy.install(Vue, {
      ...config,
      defaultContainerElement: 'body',
    })
  },
}
