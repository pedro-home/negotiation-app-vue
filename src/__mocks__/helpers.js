import { mount } from '@vue/test-utils'

export const resetDOM = () => {
  document.head.innerHTML = ''
  document.body.innerHTML = ''
}

export const render = (Component, props) => {
  resetDOM()

  return mount(Component, {
    propsData: props,
    attachToDocument: true,
  })
}
