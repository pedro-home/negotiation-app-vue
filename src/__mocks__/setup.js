import Vue from 'vue'
import Buefy from 'buefy'

// Makes sure that Jest uses buefy
Vue.use(Buefy)
