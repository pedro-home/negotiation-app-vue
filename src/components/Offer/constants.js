/**
 * Number regex to validate if a value is in a number format
 */
export const VALIDATE_NUMBER_REGEX = /^-?\d+(,\d+)*(\.\d+(e\d+)?)?$/
