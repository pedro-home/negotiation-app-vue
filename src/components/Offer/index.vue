<template>
  <div>
    <b-field>
      <b-input v-model="inputValue" :placeholder="text" @input="validate"></b-input>
    </b-field>
    <b-button :disabled="isDisabled" @click="() => onSubmit(inputValue)">Submit</b-button>
  </div>
</template>

<script>
import { VALIDATE_NUMBER_REGEX } from './constants'

export default {
  props: {
    text: {
      type: String,
      required: true,
    },
    onSubmit: {
      type: Function,
      required: true,
    },
  },
  methods: {
    validate: function() {
      // Enables submit button according to the input value validation
      this.isDisabled = !this.inputValue.match(VALIDATE_NUMBER_REGEX)
    },
  },
  data: () => ({
    inputValue: '',
    isDisabled: true,
  }),
}
</script>
