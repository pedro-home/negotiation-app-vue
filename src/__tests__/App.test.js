import { render } from '@/__mocks__/helpers'
import Offer from '../App.vue'

describe('Given <App />', () => {
  let wrapper
  let props = {}
  beforeEach(() => {
    const defaultProps = {}
    wrapper = render(Offer, { ...defaultProps, ...props })
  })

  describe('When initializes', () => {
    describe('And uses default props', () => {
      it('Then should match snapshot', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
