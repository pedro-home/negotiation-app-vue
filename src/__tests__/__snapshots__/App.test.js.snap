// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <App /> When initializes And uses default props Then should match snapshot 1`] = `
<div
  id="app"
>
  <div
    class="b-tabs is-fullwidth"
  >
    <nav
      class="tabs is-toggle is-fullwidth"
    >
      <ul>
        <li
          class="is-active"
        >
          <a>
            <!---->
             
            <span>
              Employer-Tab
            </span>
          </a>
        </li>
        <li
          class=""
        >
          <a>
            <!---->
             
            <span>
              Employee-Tab
            </span>
          </a>
        </li>
      </ul>
    </nav>
     
    <section
      class="tab-content is-transitioning"
    >
      <div
        class="tab-item v-enter v-enter-active"
        style=""
      >
        <div
          data-test="employer"
        >
          <div
            class="field"
          >
            <!---->
             
            <div
              class="control is-clearfix"
            >
              <input
                autocomplete="on"
                class="input"
                placeholder="Enter maximum salary"
                type="text"
              />
               
              <!---->
               
              <!---->
               
              <!---->
            </div>
             
            <!---->
          </div>
           
          <button
            class="button"
            disabled="disabled"
            type="button"
          >
            <!---->
             
            <span>
              Submit
            </span>
             
            <!---->
          </button>
        </div>
      </div>
       
      <div
        class="tab-item"
        style="display: none;"
      >
        <div
          data-test="employee"
        >
          <div
            class="field"
          >
            <!---->
             
            <div
              class="control is-clearfix"
            >
              <input
                autocomplete="on"
                class="input"
                placeholder="Enter minimum salary"
                type="text"
              />
               
              <!---->
               
              <!---->
               
              <!---->
            </div>
             
            <!---->
          </div>
           
          <button
            class="button"
            disabled="disabled"
            type="button"
          >
            <!---->
             
            <span>
              Submit
            </span>
             
            <!---->
          </button>
        </div>
      </div>
    </section>
  </div>
</div>
`;
